doc/usage.rst

```diff
@@ -138,6 +138,25 @@ The *key-string* is a string of 16 characters which is used to
 encrypt each file of Python byte-code before it is stored in
 the archive inside the executable file.
 
+.. _extracting libraries:
+
+Extracting Libraries
+~~~~~~~~~~~~~~~~~~~~
+
+When building an executable using ``--onefile``, |PyInstaller| generates a single
+executable file with all the required Python libraries and modules packaged
+inside it. When the executable is run, a bootloader will extract all the libraries
+into a temporary directory, and will run the application from the temporary directory.
+
+The default location for the temporary directory is the system temporary directory.
+Some operating systems allow you to configure the system temporary directory as
+non-executable, which will prevent the extracted application from executing.
+
+In order to work around that, there are two options. One is to set the environment
+variable ``TMPDIR`` to point to a location which allows executing programs. In
```

Comment:
Please refer to the section where we describe this. And name all the vars here (or none of them).